### PR TITLE
Kv keys updates

### PR DESCRIPTION
**Pull request by g2vinay**
1. Drops commons codec dependency from KV keys
2. Adds backend parsing support to read Symmetric key Stored as a secret in KV.

NOTE: No Public API changes are in this PR.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-keys/pom.xml [hunk 1]
```diff
@@ -46,12 +46,6 @@
       <version>1.1.0</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
     </dependency>
 
-    <dependency>
-      <groupId>commons-codec</groupId>
-      <artifactId>commons-codec</artifactId>
-      <version>1.13</version> <!-- {x-version-update;commons-codec:commons-codec;external_dependency} -->
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-keys/pom.xml]: should we remove this library from the external_dependencies.txt too?

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java [hunk 2]
```diff
@@ -611,8 +622,12 @@ private boolean ensureValidKeyAvailable() {
         boolean keyAvailableLocally = true;
         if (this.key == null) {
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java]: do we want a check for `keyCollection` ?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTestBase.java [hunk 3]
```diff
@@ -337,7 +337,8 @@ static void assertKeyEquals(CreateKeyOptions expected, KeyVaultKey actual) {
     public String getEndpoint() {
         final String endpoint = interceptorManager.isPlaybackMode()
                 ? "http://localhost:8080"
-                : System.getenv("AZURE_KEYVAULT_ENDPOINT");
+                : "https://cameravault.vault.azure.net";
+//                : System.getenv("AZURE_KEYVAULT_ENDPOINT");
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTestBase.java]: remove vault end point url here and below

Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTestBase.java]: updated

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java (hunk 2 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java]: added the check

File: sdk/keyvault/azure-security-keyvault-keys/pom.xml (hunk 1 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-keys/pom.xml]: need to check if any other SDK still has this dependency.
Best to clean that up in a follow up PR.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java [hunk 4]
```diff
@@ -55,11 +55,13 @@
 @ServiceClient(builder = CryptographyClientBuilder.class, isAsync = true, serviceInterfaces = CryptographyService.class)
 public class CryptographyAsyncClient {
     static final String KEY_VAULT_SCOPE = "https://vault.azure.net/.default";
+    public static final String SECRETS_COLLECTION = "secrets";
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java]: Your PR comment says there are no public API changes, but this is a new public API. Is it required?

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SignatureEncoding.java [hunk 5]
```diff
@@ -86,8 +87,21 @@ private SignatureEncoding() { }
...
+                ex.getMessage() + " " + Arrays.toString(encodeHex(signature, HEX_LOWER))).initCause(ex);
+        }
+    }
+
+    private static char[] encodeHex(byte[] data, char[] toDigits) {
+        int l = data.length;
+        char[] out = new char[l << 1];
+        int i = 0;
+
+        for (int var5 = 0; i < l; ++i) {
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SignatureEncoding.java]: Why is this called `var5`?

Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SignatureEncoding.java]: renamed it to j.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java (hunk 4 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java]: yeah, that's not needed.
made it private.

**Pull request merged**